src/standalone/Vegetation/photosynthesis_farquhar.jl

```diff
@@ -0,0 +1,716 @@
+        :CO2_compensation_point_25c => :Γstar25,
+        :Farquhar_curvature_parameter => :θj,
+        :kelvin_25C => :To,
+        :photosystem_II_quantum_yield => :ϕ,
+        :O2_michaelis_menten => :Ko25,
+        :CO2_michaelis_menten => :Kc25,
+        :dark_respiration_factor => :fC3,
+        :O2_activation_energy => :ΔHko,
+        :low_water_pressure_sensitivity => :sc,
+        :Rd_activation_energy => :ΔHRd,
+        :Vcmax_activation_energy => :ΔHVcmax,
+        :Γstar_activation_energy => :ΔHΓstar,
+        :CO2_activation_energy => :ΔHkc,
+        :moisture_stress_ref_water_pressure => :pc,
+    )
+    parameters = CP.get_parameter_values(toml_dict, name_map, "Land")
+    FT = CP.float_type(toml_dict)
+    C4_parameters = (
+        :fC4 => FT(0.025),
+        :Q10 => FT(2),
+        :s1 => FT(0.3),
+        :s2 => FT(313.15),
+        :s3 => FT(0.2),
+        :s4 => FT(288.15),
+        :s5 => FT(1.3),
+        :s6 => FT(328.15),
+        :E => FT(0.05),
+    )
+    # if is_c3 is a field, is_c3 may contain values between 0.0 and 1.0 after regridding
+    # this deals with that possibility by rounding to the closest int
+    is_c3 = max.(min.(is_c3, FT(1)), FT(0)) # placeholder
+    is_c3 = round.(is_c3)
+    MECH = typeof(is_c3)
+    Vcmax25 = FT.(Vcmax25)
+    VC = typeof(Vcmax25)
+    return FarquharParameters{FT, MECH, VC}(;
+        is_c3,
+        Vcmax25,
+        parameters...,
+        C4_parameters...,
+        kwargs...,
+    )
+end
+
+"""
+    max_electron_transport_farquhar(
+        Vcmax25::FT,
+        ΔHJmax::FT,
+        T::FT,
+        To::FT,
+        R::FT,
+    ) where {FT}
+
+Computes the maximum potential rate of electron transport (`Jmax`),
+in units of mol/m^2/s,
+as a function of Vcmax at 25 °C (`Vcmax25`),
+a constant (`ΔHJmax`), a standard temperature (`To`),
+the unversal gas constant (`R`), and the temperature (`T`).
+
+See Table 11.5 of G. Bonan's textbook,
+Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function max_electron_transport_farquhar(
+    Vcmax25::FT,
+    ΔHJmax::FT,
+    T::FT,
+    To::FT,
+    R::FT,
+) where {FT}
+    Jmax25 = Vcmax25 * FT(exp(1))
+    Jmax = Jmax25 * arrhenius_function(T, To, R, ΔHJmax)
+    return Jmax
+end
+
+"""
+    electron_transport_farquhar(APAR::FT,
+                       Jmax::FT,
+                       θj::FT,
+                       ϕ::FT) where {FT}
+
+Computes the rate of electron transport (`J`),
+in units of mol/m^2/s, as a function of
+the maximum potential rate of electron transport (`Jmax`),
+absorbed photosynthetically active radiation (`APAR`),
+an empirical "curvature parameter" (`θj`; Bonan Eqn 11.21)
+and the quantum yield of photosystem II (`ϕ`).
+
+See Ch 11, G. Bonan's textbook, Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function electron_transport_farquhar(
+    APAR::FT,
+    Jmax::FT,
+    θj::FT,
+    ϕ::FT,
+) where {FT}
+    # Light utilization of APAR
+    IPSII = ϕ * APAR / 2
```

Comment:
Same here, add a comment to say APAR can be leaf (farquhar) or canopy (pmodel), also explain in docstring